providers/harness/pkg/provider.go

```diff
@@ -0,0 +1,310 @@
+package harness
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/harness/ff-golang-server-sdk/client"
+	harness "github.com/harness/ff-golang-server-sdk/client"
+	"github.com/harness/ff-golang-server-sdk/evaluation"
+	"github.com/harness/ff-golang-server-sdk/types"
+	of "github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+const providerNotReady = "Provider not ready"
+const generalError = "general error"
+
+type Provider struct {
+	providerConfig ProviderConfig
+	harnessClient  *client.CfClient
```

Comment:
@davejohnston is there any risk having multiple clients in a runtime? Do they share any global state? Some SDKs don't play well with that. I suspect not, just pointing it out.

Reply:
@toddbaert no we should be good here.  Not all our SDKs behave well but golang is one that we have initialised multiple instances in a single application.

Reply:
@toddbaert trying to understand, why would this can cause having multiple clients in a runtime? can you please elaborate ?

Reply:
Some vendors assume only one SDK client (not openfeature client) exists in the runtime. They share some global state in some cases.

According to @davejohnston this isn't an issue here, at least with Harness' golang SDK.